src/Twig/Components/SidebarSubscriptionComponent.php

```diff
@@ -25,22 +26,22 @@ class SidebarSubscriptionComponent
 
     public ?string $sort;
 
+    public function __construct(private readonly MagazineRepository $magazineRepository)
+    {
+    }
+
     #[PostMount]
     public function PostMount(): void
     {
         $max = 50;
         $this->magazines = [];
-        foreach ($this->user->subscriptions as /* @type MagazineSubscription $sub */ $sub) {
-            $this->magazines[] = $sub->magazine;
-        }
         if (ThemeSettingsController::ALPHABETICALLY === $this->sort) {
-            usort($this->magazines, fn ($a, $b) => $a->name > $b->name ? 1 : -1);
+            $this->magazines = $this->magazineRepository->findMagazineSubscriptionsOfUser($this->user, SubscriptionSort::Alphabetically, $max);
         } else {
-            usort($this->magazines, fn ($a, $b) => $a->lastActive < $b->lastActive ? 1 : -1);
+            $this->magazines = $this->magazineRepository->findMagazineSubscriptionsOfUser($this->user, SubscriptionSort::LastActive, $max);
         }
-        if (\sizeof($this->magazines) > $max) {
+        if (\sizeof($this->magazines) === $max) {
             $this->tooManyMagazines = true;
         }
-        $this->magazines = \array_slice($this->magazines, 0, $max);
```

Comment:
yea this of course was indeed not ideal...  lol

Reply:
yep. Fetching **all** magazines, then sorting them and then discarding all but 50 of them is really not ideal performance wise

Reply:
Not exactly.. In fact first fetching all magazines by looping over all elements and adding them manually to an array is also less than ideal xD